examples/cndpfwd/parse-args.c

```diff
@@ -77,16 +77,27 @@ process_callback(jcfg_info_t *j __cne_unused, void *_obj, void *arg, int idx)
                     CNE_ERR_RET("UDS handshake failed\n");
             }
         } else if (!strncmp(obj.opt->name, FIB_RULES_TAG, nlen)) {
-            uint16_t i;
             if (obj.opt->val.type == ARRAY_OPT_TYPE) {
                 f->fib_rules = calloc(obj.opt->val.array_sz, sizeof(char *));
                 if (!f->fib_rules)
                     CNE_ERR_RET("Unable to allocate fib_rules array\n");
 
-                for (i = 0; i < obj.opt->val.array_sz; ++i)
+                for (int i = 0; i < obj.opt->val.array_sz; ++i) {
                     f->fib_rules[i] = obj.opt->val.arr[i]->str;
-
-                f->fib_size = i;
+                    f->fib_size++;
+                }
+            }
+        } else if (!strncmp(obj.opt->name, HS_PATTERN_TAG, nlen)) {
+            if (obj.opt->val.type == ARRAY_OPT_TYPE) {
+                f->hs_patterns = calloc(obj.opt->val.array_sz, sizeof(char *));
+                if (!f->hs_patterns)
+                    CNE_ERR_RET("Unable to allocate hsfwd information array\n");
```

Comment:
Silly question - but should we check if umem memory has been mmap_alloc'd or if pd below has been alloced (or any other malloc'd variables) and do some cleanup before we return in error from the OPTIONS config processing?

Reply:
Yes, I agree we should be doing a better job of cleaning up. I can add some code to cleanup the malloc'd memory in the new code, but we need to do a much better job of freeing memory everyplace or just reply on the OS to cleanup.

Reply:
yeah that would work